src/lib/gadgets/bitwise.ts

```diff
@@ -0,0 +1,131 @@
+import { Provable } from '../provable.js';
+import { Field as Fp } from '../../provable/field-bigint.js';
+import { Field } from '../field.js';
+import * as Gates from '../gates.js';
+
+export { xor };
+
+function xor(a: Field, b: Field, length: number) {
+  // check that both input lengths are positive
+  assert(length > 0, `Input lengths need to be positive values.`);
+
+  // check that length does not exceed maximum field size in bits
+  assert(
```

Comment:
~I wonder if you might also want to check that the given inputs fit in the desired length and fail otherwise?~ Never mind, I saw this is checked inside the case where `a` and `b` are constants below.